### PR TITLE
Implement support for passing a .env file to corerun to easily set environment variables for stress testing.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Introduce support into corerun for providing a "dotenv" file based on the format supported by [python-dotenv](https://pypi.org/project/python-dotenv/) that corerun will use to set environment variables before loading CoreCLR.

This can be used to set various different stress modes by just editing a text file with a single format instead of having to edit a script which you then have to run.

The file can be provided to corerun with the `-e` or `--env` flags.

For example, diagnosing #57915 took setting and regularly changing 5+ environment variables, which was error prone.

TODO:

- [x] Docs
- [x] Update the bash/batch scripts for tests to support dotenv files.
- [ ] Investigation into updating "scenario" Helix runs for corerun-based testing to use this.

### CODE REVIEW

**Review by AaronRobinsonMSFT — COMMENTED** [review]
We should also update the doc on `corerun` usage - https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/docs/workflow/testing/using-corerun.md

**Comment by jkoritzinsky** [comment]
Only test failure is the xunit test harness collection issue.

File: docs/workflow/testing/using-corerun.md [hunk 1]
```diff
@@ -98,3 +98,6 @@ See `corerun --help` for additional details.
 
 `--debug` - Wait for a debugger to attach prior to loading the .NET runtime.
 - For example, `corerun --debug HelloWorld.dll`
+
+`--env` - Pass the path to a file in the `dotenv` format to specify environment variables for the test run.
```
Comment (AaronRobinsonMSFT) [inline docs/workflow/testing/using-corerun.md:102]:
```suggestion
`--env` - Pass the path to a file in the [`dotenv`](https://github.com/motdotla/dotenv) format to specify environment variables for the test run.
```

File: src/coreclr/hosts/corerun/dotenv.cpp [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+            break;
+        }
+
+        if (contents.eof())
+        {
+            _environmentVariables = {};
+            break;
+        }
+
+        if (!read_var_value(contents, [&](std::string name)
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/coreclr/hosts/corerun/dotenv.cpp:318]:
```suggestion
        // Handle variable expansion scenarios
        if (!read_var_value(contents, [&](std::string name)
```

File: src/coreclr/hosts/corerun/dotenv.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+public:
+    dotenv() = default;
+    dotenv(pal::string_t dotEnvFilePath, std::istream& contents);
+    void load_into_current_process() const;
+
+    static void self_test();
+};
+
+#endif
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/hosts/corerun/dotenv.hpp:28]:
```suggestion
#endif // __DOTENV_HPP__
```

**Review by AaronRobinsonMSFT — APPROVED** [review]
Thanks!

**Pull request merged**